dotnet/samples/Concepts/ChatCompletion/MistralAI_FunctionCalling.cs

```diff
@@ -0,0 +1,202 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.ComponentModel;
+using System.Text.Json.Serialization;
+using Microsoft.OpenApi.Extensions;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Connectors.MistralAI;
+
+namespace ChatCompletion;
+
+/// <summary>
+/// Demonstrates the use of function calling with MistralAI.
+/// </summary>
+public sealed class MistralAI_FunctionCalling(ITestOutputHelper output) : BaseTest(output)
+{
+    [Fact]
+    public async Task AutoInvokeKernelFunctionsAsync()
+    {
+        // Create a logging handler to output HTTP requests and responses
+        var handler = new LoggingHandler(new HttpClientHandler(), this.Output);
+        HttpClient httpClient = new(handler);
+
+        // Create a kernel with MistralAI chat completion and WeatherPlugin
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.AddMistralChatCompletion(
+                modelId: TestConfiguration.MistralAI.ChatModelId!,
+                apiKey: TestConfiguration.MistralAI.ApiKey!,
+                httpClient: httpClient);
+        kernelBuilder.Plugins.AddFromType<WeatherPlugin>();
+        Kernel kernel = kernelBuilder.Build();
+
+        // Invoke chat prompt with auto invocation of functions enabled
+        const string ChatPrompt = @"
+            <message role=""user"">What is the weather like in Paris?</message>
+        ";
+        var executionSettings = new MistralAIPromptExecutionSettings { ToolCallBehavior = MistralAIToolCallBehavior.AutoInvokeKernelFunctions };
+        var chatSemanticFunction = kernel.CreateFunctionFromPrompt(
+            ChatPrompt, executionSettings);
+        var chatPromptResult = await kernel.InvokeAsync(chatSemanticFunction);
+
+        Console.WriteLine(chatPromptResult);
+    }
+
+    [Fact]
+    public async Task AutoInvokeKernelFunctionsMultipleCallsAsync()
+    {
+        // Create a logging handler to output HTTP requests and responses
+        var handler = new LoggingHandler(new HttpClientHandler(), this.Output);
+        HttpClient httpClient = new(handler);
+
+        // Create a kernel with MistralAI chat completion and WeatherPlugin
+        IKernelBuilder kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.AddMistralChatCompletion(
+                modelId: TestConfiguration.MistralAI.ChatModelId!,
+                apiKey: TestConfiguration.MistralAI.ApiKey!,
+                httpClient: httpClient);
```

Comment:
Extract this to be a private method and reutilize it on the other tests.

May consider doing same for the handler + httpClient.